torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -571,6 +574,56 @@ def test_backend_group(self):
         def test_backend_full_group(self):
             self._test_group_override_backend(self._init_full_group_test)
 
+        # NCCL Batch SEND RECV
+        @skip_if_no_gpu
+        @unittest.skipIf(BACKEND != "nccl", "NCCL Send Recv Only")
+        def test_batch_send_recv_nccl(self):
+            self._barrier()
+            rank = dist.get_rank()
+            rank_to_GPU = self._init_multigpu_helper()
+
+            send_recv_op_list = []
+            send_recv_tensor_list = []
+            peer_list = []
+            expected_tensor_list = []
+
+            for src in range(0, dist.get_world_size()):
+                if src == rank:
+                    continue
+                send_tensor = _build_tensor(1, rank=rank, cuda=True, rank_to_GPU=rank_to_GPU)
+                recv_tensor = _build_tensor(1, rank=rank, cuda=True, rank_to_GPU=rank_to_GPU)
+                send_recv_op_list.extend([dist.irecv, dist.isend])
+                send_recv_tensor_list.extend([recv_tensor, send_tensor])
+                peer_list.extend([src, src])
+
+            reqs = dist.batch_isend_irecv(send_recv_op_list, send_recv_tensor_list, peer_list)
+            for req in reqs:
+                req.wait()
+            self._barrier()
+
+        @skip_if_no_gpu
+        @unittest.skipIf(BACKEND != "nccl", "NCCL Send Recv Only")
+        def test_send_recv_nccl(self):
```

Comment:
Can we test error scenarios for `batch_isend_irecv`? Some examples:

1. Invalid ops.
2. Invalid peer_list, group_list, tag_list etc. lengths.
3. Invalid mix of nccl and gloo backends.
4. Invalid set of tensors (ex: CPU tensors, sparse tensors etc.)